server/api/auth_handler.py

```diff
@@ -8,27 +8,28 @@
 auth_handler = Blueprint("auth_handler", __name__)
 
 
+def token_getter():
+    try:
+        auth_header = request.cookies.get("Authentication token")
+        if auth_header:
+            auth_token = auth_header
+            auth_token = User.decode_auth_token(auth_token)
+            return auth_token
+        else:
+            raise Exception
+    except:
+        responseObject = {
+            "status": "fail",
+            "message": "User is not logged in."
+        }
+    return make_response(jsonify(responseObject)), 401
+
+
 # Wrapper that obtains and verifies an auth token in a cookie. Returns true if valid, false otherwise.
 def login_cookie_getter(f):
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        try:
-            auth_header = request.cookies.get("Authentication token")
-            if auth_header:
-                auth_token = auth_header
-                auth_token = User.decode_auth_token(auth_token)
-                if not isinstance(auth_token, str):
-                    g.user = User.query.filter_by(id=auth_token).first()
-                else:
-                    raise Exception
-            else:
-                raise Exception
-        except:
-            responseObject = {
-                "status": "fail",
-                "message": "User is not logged in."
-            }
-            return make_response(jsonify(responseObject)), 401
+        token_getter()
```

Comment:
I like what you've done with the token_getter, but there's a change that may need to be made to preserve the purpose of the login cookie_getter. I think that token_getter() should be replaced by:

```
auth_token = token_getter()
try:
    g.user = User.query.filter_by(id=auth_token).first()
except:
    raise Exception
```

This will set g.user to the current user, which is important for the User API.